### PR TITLE
Setup CI.

### PR DESCRIPTION
**Pull request by arielshaqed**
Use Cirrus-CI (free tier for public repo).

### CODE REVIEW

**Comment by arielshaqed** [comment]
Testing: Should take effect _on this branch_.

File: .cirrus.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+container:
+  image: node:latest
+
+test_task:
+  node_modules_cache:
+    folder: node_modules
+    compile_script: tsc
```
Comment (fkorotkov) [inline .cirrus.yml]: `compile_script` should be one level up (outside `node_modules_cache`)

Comment (arielshaqed) [reply .cirrus.yml]: Thanks, that makes sense and works!  I was wondering why it never actually ran on Cirrus.

Also Best Product Support *EVER* :-) !

**Comment by arielshaqed** [comment]
@michaeladda sez:
> did you add package-lock.json on purpose?

Yes.

1. I want to use `npm ci` on Cirrus, this is their recommended task.
2. It's an external package, users deserve to know what version got tested.  They might not have continuous integration set up to test with whatever semanti^Wrandom version matched `package.lock` today.

**Pull request merged**